compiler/compiler-settings.cpp

```diff
@@ -283,19 +290,23 @@ void CompilerSettings::init() {
   if (!no_pch.get()) {
     ss << " -Winvalid-pch -fpch-preprocess";
   }
-  if (dynamic_incremental_linkage.get()) {
+  if (mode.get() == "k2-component" ||  dynamic_incremental_linkage.get()) {
     ss << " -fPIC";
   }
   if (vk::contains(cxx.get(), "clang")) {
     ss << " -Wno-invalid-source-encoding";
+  } else {
+    if (mode.get() == "k2-component") {
+      ss << " -fcoroutines";
+    }
```

Comment:
I think we can remove it because `cxx.get()` must contain clang substring in case of `k2` mode.

Reply:
fix